wallet/src/main/scala/org/bitcoins/wallet/config/WalletAppConfig.scala

```diff
@@ -7,14 +7,26 @@ import org.bitcoins.wallet.db.WalletDbManagement
 import scala.util.Failure
 import scala.util.Success
 import java.nio.file.Files
+import org.bitcoins.core.hd.HDPurpose
+import org.bitcoins.core.hd.HDPurposes
 
-case class WalletAppConfig(conf: Config*) extends AppConfig {
+case class WalletAppConfig(private val conf: Config*) extends AppConfig {
   override val configOverrides: List[Config] = conf.toList
   override def moduleName: String = "wallet"
   override type ConfigType = WalletAppConfig
-  override def newConfigOfType(configs: List[Config]): WalletAppConfig =
+  override def newConfigOfType(configs: Seq[Config]): WalletAppConfig =
     WalletAppConfig(configs: _*)
 
+  lazy val defaultAccountKind: HDPurpose =
```

Comment:
so since configs can be changed dynamically, don't we want this to be a `def` so the `defaultAccountKind` can be changed as well? 

i.e. if someone changes the config to change the default account, this value with be cached and not changed. In general we should be using `def` always with config stuff so everything gets modified at runtime

Reply:
That's a good point. Right now I haven't put much though into what configuration is dynamic and what is set at initialization. Another question apart from what it is right now is what is _desired_

Pros:
- it's flexible

Cons: 
- it's global, mutable state on steroids

I think I lean more towards an approach where you have to explicitly call a method `updateConfiguration`, or something along those lines.

Reply:
Let's file an issue and worry about it later?